### PR TITLE
Blog post of datafusion-python updates 40.1.0

### PR DESCRIPTION
**Pull request by timsaucer**
Per request. Dates will need to be changed with release 40.0 comes out

### CODE REVIEW

**Comment by timsaucer** [comment]
Some other things that probably make sense to add are:

- An overview of the major new features from datafusion since we have the upgrade to 40.0 and a link to the blog post and change log for it
- Example of how `named_struct` is cleaner

File: _posts/2024-07-22-python-datafusion-40.0.0.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+---
+layout: post
+title: "Apache DataFusion Python 40.0.0 Released, Significant usability updates"
+date: "2024-08-13 00:00:00"
```
Comment (andygrove) [inline _posts/2024-07-22-python-datafusion-40.0.0.md:4]:
```suggestion
title: "Apache DataFusion Python 40.1.0 Released, Significant usability updates"
date: "2024-08-20 00:00:00"
```

File: _posts/2024-07-22-python-datafusion-40.0.0.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+## Introduction
+
+We are happy to announce that [DataFusion in Python 40.0.0] has been released. In addition to
```
Comment (andygrove) [inline _posts/2024-07-22-python-datafusion-40.0.0.md:29]:
```suggestion
We are happy to announce that [DataFusion in Python 40.1.0] has been released. In addition to
```

File: _posts/2024-07-22-python-datafusion-40.0.0.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+This release aims to bring a better user experience to the DataFusion Python community.
+
+## What's Changed
+
+The most significant difference is that we have added wrapper functions and classes for most of the
+user facing interface. These wrappers, written in Python, contain both documentation and type
+annotations.
+
+This documenation is now available on the [DataFusion in Python](https://datafusion.apache.org/python)
```
Comment (andygrove) [inline _posts/2024-07-22-python-datafusion-40.0.0.md:60]: Would it be better to link directly to the API reference? https://datafusion.apache.org/python/api.html

**Review by andygrove — APPROVED** [review]
LGTM once the version numbers are updated. Thanks @timsaucer

File: _posts/2024-08-20-python-datafusion-40.0.0.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+Additionally, in the next release of DataFusion there have been improvements made to the user-defined
+aggregate and window functions to make them easier to use. We plan on
+[bringing these enhancements](https://github.com/apache/datafusion-python/issues/780) to this project.
+
+## Thank You
+
+We would like to thank the following members for their very helpful discussions regarding these
+updates: [@andygrove], [@max-muoto], [@slyons], [@Throne3d], [@Michael-J-Ward], [@datapythonista],
+[@austin362667], [@kylebarron], [@simicd]. The [primary PR (#750)] that includes these updates
+had and extensive conversation, leading to an significantly improved end product. Again, thank you
```
Comment (andygrove) [inline _posts/2024-08-20-python-datafusion-40.0.0.md:195]:
```suggestion
had an extensive conversation, leading to a significantly improved end product. Again, thank you
```

File: _posts/2024-08-20-python-datafusion-40.0.0.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+to all who provided input!
+
+We would like to give an special thank you to [@3ok] who created the initial version of the wrapper
+definitions. The work they did was time consuming and required exception attention to detail. It
```
Comment (andygrove) [inline _posts/2024-08-20-python-datafusion-40.0.0.md:199]:
```suggestion
definitions. The work they did was time consuming and required exceptional attention to detail. It
```

**Pull request merged**